lib/dataset.js

```diff
@@ -121,75 +127,74 @@ Dataset.prototype.rightOuterJoin = function (other) {
 };
 
 Dataset.prototype.distinct = function () {
-  return this.map(function (e) {return [e, null];})
-    .reduceByKey(function (a) {return a;}, null)
-    .map(function (a) {return a[0];});
+  return this.map(e => [e, null])
+    .reduceByKey(a => a, null)
+    .map(a => a[0]);
 };
 
 Dataset.prototype.intersection = function (other) {
-  function mapper(e) {return [e, 0];}
-  function reducer(a) {return a + 1;}
-  var a = this.map(mapper).reduceByKey(reducer, 0);
-  var b = other.map(mapper).reduceByKey(reducer, 0);
-  return a.coGroup(b).flatMap(function (a) {
+  const mapper = e => [e, 0];
+  const reducer = a => a + 1;
+  const a = this.map(mapper).reduceByKey(reducer, 0);
+  const b = other.map(mapper).reduceByKey(reducer, 0);
+  return a.coGroup(b).flatMap(a => {
     return (a[1][0].length && a[1][1].length) ? [a[0]] : [];
   });
 };
 
 Dataset.prototype.subtract = function (other) {
-  function mapper(e) {return [e, 0];}
-  function reducer(a) {return a + 1;}
-  var a = this.map(mapper).reduceByKey(reducer, 0);
-  var b = other.map(mapper).reduceByKey(reducer, 0);
-  return a.coGroup(b).flatMap(function (a) {
-    var res = [];
+  const mapper = e => [e, 0];
+  const reducer = a => a + 1;
+  const a = this.map(mapper).reduceByKey(reducer, 0);
+  const b = other.map(mapper).reduceByKey(reducer, 0);
+  return a.coGroup(b).flatMap(a => {
+    const res = [];
     if (a[1][0].length && (a[1][1].length === 0))
       for (var i = 0; i < a[1][0][0]; i++) res.push(a[0]);
     return res;
   });
 };
 
-Dataset.prototype.keys = function () {return this.map(function (a) {return a[0];});};
+Dataset.prototype.keys = function () {return this.map(a => a[0]);};
 
-Dataset.prototype.values = function () {return this.map(function (a) {return a[1];});};
+Dataset.prototype.values = function () {return this.map(a => a[1]);};
 
 Dataset.prototype.lookup = thenify(function (key, done) {
-  return this.filter(function (kv, args) {return kv[0] === args.key;}, {key: key})
-    .map(function (kv) {return kv[1];}).collect(done);
+  return this.filter((kv, args) => kv[0] === args.key, {key})
+    .map(kv => kv[1])
+    .collect(done);
 });
 
 Dataset.prototype.countByValue = thenify(function (done) {
-  return this.map(function (e) {return [e, 1];})
-    .reduceByKey(function (a, b) {return a + b;}, 0)
+  return this.map(e => [e, 1])
+    .reduceByKey((a, b) => a + b, 0)
     .collect(done);
 });
 
 Dataset.prototype.countByKey = thenify(function (done) {
-  return this.mapValues(function () {return 1;})
-    .reduceByKey(function (a, b) {return a + b;}, 0)
+  return this.mapValues(a => 1) // eslint-disable-line no-unused-vars
```

Comment:
should be: `return this.mapValues(() => 1);`

Reply:
This was throwing a bunch of errors in the test suit for some odd reason because you are right that should be fully valid. I'll try it again and put the error up here.

Reply:
This seems to cause a scope issue, which seems weird that have a named argument wouldn't. I can put it back to `function () {return 1;})` for now if that is preferred over the eslint disable.

```
dataset.js:1142
    tmp[i] = [data[i][0], this.mapper(data[i][1], this.args, task)];
                               ^

TypeError: this.mapper is not a function
```